### PR TITLE
test: assert clean close in win11 smoke harness

### PR DESCRIPTION
**Pull request by amanthanvi**
## Summary

- assert clean WM_CLOSE shutdown in the interactive Win11 smoke harness
- fail the smoke run if the app never exposes a main window handle or exits non-zero after close
- clarify repo AGENTS guidance to target only the manthanvi/winghostty fork for PR work, never upstream Ghostty

## Validation

- [x] powershell -ExecutionPolicy Bypass -File test/windows/interactive-win11-smoke.ps1 -Rebuild -ResetState
- [x] cmd /c scripts\dev-windows.cmd zig build test -Dtest-filter=""win32 shouldUseIntegratedTitlebar gates on Win11 build floor""
- [x] git diff --check -- AGENTS.md test/windows/interactive-win11-smoke.ps1

## Risks / Follow-ups

- this only hardens the validation harness; it does not change the Win32 runtime path itself
- bot review triggers depend on which GitHub apps are installed and listening on this fork

## Summary by Sourcery

Strengthen the Windows 11 interactive smoke test harness to validate clean GUI shutdown behavior and update agent contribution guidelines for this fork.

Documentation:
- Clarify AGENTS guidance to only open PRs against the amanthanvi/winghostty fork and to refuse upstream publication requests.

Tests:
- Extend the interactive Windows 11 smoke test to verify the app exposes a main window, sends WM_CLOSE, and requires a clean, zero-exit shutdown for success.

## Summary by CodeRabbit

* **Documentation**
  * Updated agent behavior guidance with fork-specific restrictions and upstream publishing constraints.
  * Added self-correction documentation for PowerShell exit-code handling edge case with native API fallback.

* **Tests**
  * Enhanced Windows shutdown validation testing with WM_CLOSE messaging verification and native exit-code fallback handling.

### CODE REVIEW

**Comment by amanthanvi** [comment]
@copilot Please review this PR for correctness and Windows edge cases.

@coderabbitai review

@sourcery-ai review

@greptileai

**Comment by amanthanvi** [comment]
Codex review summary:

- inspected only the scoped 2-file diff (AGENTS.md and test/windows/interactive-win11-smoke.ps1)
- no blocking issues found in the scoped change
- the harness update is limited to WM_CLOSE to process-exit-0 validation and does not alter the Win32 runtime path itself
- local validation passed before PR creation

**Comment by amanthanvi** [comment]
Addressed the review feedback on the smoke harness:\n- switched WM_CLOSE delivery to non-blocking PostMessageW\n- fail early if the app exits before exposing MainWindowHandle during close validation\n- tied the close wait to TimeoutSeconds\n- prefer Process.ExitCode, with a native GetExitCodeProcess fallback for the dev-windows bootstrap path where the managed property is null in practice\n\n@codex review\n@copilot review\n@coderabbitai review\n@sourcery-ai review\n@greptileai

File: test/windows/interactive-win11-smoke.ps1 [hunk 1]
```diff
@@ -84,12 +100,52 @@ try {
...
+                if (-not [InteractiveWin11SmokeNative]::GetExitCodeProcess($processHandle, [ref] $exitCode)) {
+                    $failureReason = "winghostty exited after WM_CLOSE but GetExitCodeProcess failed: $([Runtime.InteropServices.Marshal]::GetLastWin32Error())"
+                }
+                elseif ($exitCode -ne 0) {
+                    $failureReason = "winghostty exited after WM_CLOSE with exit code $exitCode"
+                }
+                else {
+                    $closePassed = $true
+                }
+            }
```
[the remark this answers is not in the document]
Comment (amanthanvi) [reply test/windows/interactive-win11-smoke.ps1:158]: Keeping the native fallback intentionally. I reproduced this under the exact \\scripts/dev-windows.cmd\\ bootstrap path with a live \\winghostty.exe\\ child: after \\WaitForExit()\\ and \\Refresh()\\, \\$process.ExitCode\\ is still null there, even though the process exited cleanly. The harness now prefers \\Process.ExitCode\\ first and only falls back to \\GetExitCodeProcess\\ when the managed property is unavailable in that environment. I also recorded the trap in \\AGENTS.md\\ so we do not regress back to the null-exit-code path.

**Comment by amanthanvi** [comment]
@copilot Please re-review the current head commit da4d3496; the earlier WM_CLOSE and MainWindowHandle findings were addressed, local harness validation reran cleanly, and the PR checks are now green.
